dataflow/src/main/java/org/checkerframework/dataflow/cfg/builder/CFGTranslationPhaseOne.java

```diff
@@ -1311,6 +1313,20 @@ protected List<Node> convertCallArguments(
             } else {
                 assert lastParamType instanceof ArrayType
                         : "variable argument formal must be an array";
+                // Handle anonymous constructors that extend a class with an enclosing type.
```

Comment:
Don't add comments that explain the history, explain the code as it is now.
The reference to AnnotatedTypes is confusing now, as there is no code to look at. It's not borrowed, it's moved from there to here.

Reply:
Thanks! I reworeded the comments.